### PR TITLE
Finished Basic Itinerary Feature

### PR DESCRIPTION
**Pull request by Sukrut1881**
1) Now u can create ur own Itinerary from scratch using : 

newItinerary 23/04/15 26/04/15 MBS testList 1 /venue Simei /do dancing /and singing /and swinging 2 /venue Bedok /do running /and jumping /and swinging

(exceptions have been added) 

2) Can list your saved itineraries by name using:

listItinerary

3) Can view any of your saved Itineraries by using : 

showItinerary <itinerary number>

4) Can ask SGTravel for Recommendations using : 

recommend <location> between <start> and <end>

5) Save the recommended list using : 

addThisList command

6) Added Tests for every new command added 

Things to be done:

1) Rename/Refactoring
2) More tests (need to add for Agenda, Itinerary and Storage)
3) Break down large functions
4) Possibly break down Storage into a few classes
5) Implement PDF with showItinerary
6) Integrate with HC's feature
7) Change all instances of List<Agenda,> to AgendaList

### CODE REVIEW

File: src/main/java/duke/logic/parsers/ParserUtil.java [hunk 1]
```diff
@@ -30,6 +53,130 @@ protected static Todo createTodo(String userInput) throws DukeException {
         return new Todo(description);
     }
 
+    /**
+     * Parses the userInput and return a new Event constructed from it.
+     *
+     * @param userInput The userInput read by the user interface.
+     * @return The new Event object.
+     */
+    protected static Event createEvent(String userInput) throws DukeException {
```
Comment (hongchuan97) [inline src/main/java/duke/logic/parsers/ParserUtil.java]: i throw a few of this methods to the parser itself already cause theres no other uses for it

File: src/main/java/duke/model/ModelManager.java [hunk 2]
```diff
@@ -105,6 +110,49 @@ public RouteManager getRouteManager() {
         return routeManager;
     }
 
+    /**
```
Comment (hongchuan97) [inline src/main/java/duke/model/ModelManager.java]: All this methods shold be in storage

File: src/main/resources/data/recommendations.txt [hunk 3]
```diff
@@ -1,32 +1,32 @@
 SENTOSA | 1.2498144130047 | 103.82948052356899 | 27572.8505303748 | 25823.6513112113
```
Comment (hongchuan97) [inline src/main/resources/data/recommendations.txt]: gitignore this file

File: src/main/java/duke/logic/parsers/Parser.java [hunk 4]
```diff
@@ -8,6 +8,20 @@
 import duke.logic.commands.ExitCommand;
 import duke.logic.commands.HelpCommand;
 import duke.logic.commands.ListCommand;
+import duke.logic.commands.ListItineraryCommand;
+import duke.logic.commands.LocationSearchCommand;
+import duke.logic.commands.MarkDoneCommand;
```
Comment (hongchuan97) [inline src/main/java/duke/logic/parsers/Parser.java]: unused imports if i remember correctly

File: src/main/java/duke/logic/parsers/ParserUtil.java (hunk 1 above)
Comment (Sukrut1881) [reply src/main/java/duke/logic/parsers/ParserUtil.java]: Ok then i will delete the duplicate ones

File: src/main/java/duke/logic/commands/AddSampleItineraryCommand.java [hunk 5]
```diff
@@ -29,8 +31,11 @@ public AddSampleItineraryCommand() throws DukeDateTimeParseException, FileLoadFa
      * @param model The model object containing information about the user.
      */
     @Override
-    public CommandResultText execute(Model model) {
+    public CommandResultText execute(Model model) throws DukeException, FileNotFoundException {
         // Add to the list of Itineraries
+        this.itinerary = model.readRecommendations();
```
Comment (Jefferson111) [inline src/main/java/duke/logic/commands/AddSampleItineraryCommand.java]: I think u can localised itinerary

File: src/main/java/duke/logic/commands/ListItineraryCommand.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import duke.logic.commands.results.CommandResultText;
+import duke.model.Model;
+
+import java.io.FileNotFoundException;
+
+/**
+ * Creates a new custom itinerary.
+ */
+public class ListItineraryCommand extends Command {
+
```
Comment (Jefferson111) [inline src/main/java/duke/logic/commands/ListItineraryCommand.java]: constants

File: src/main/java/duke/logic/parsers/ParserUtil.java [hunk 7]
```diff
@@ -118,7 +264,59 @@ public static int getIntegerIndexInList(int index, int listSize, String userInpu
...
         throw new QueryOutOfBoundsException("INTEGER");
     }
+
+    /**
+     * Gets the integer index at index in a String list delimited by whitespace.
+     *
+     * @param userInput The users entered command which contains details of the Itinerary.
+     * @return The Itinerary object created.
+     */
+    public static Itinerary createNewItinerary(String userInput) throws DukeException {
```
Comment (Jefferson111) [inline src/main/java/duke/logic/parsers/ParserUtil.java]: refactor

File: src/main/java/duke/logic/commands/NewItineraryCommand.java [hunk 8]
```diff
@@ -3,19 +3,33 @@
...
 
     /**
      * Executes this command on the given task list and user interface.
      *
      * @param model The model object containing information about the user.
      */
     @Override
-    public CommandResultText execute(Model model) throws DukeException {
-        return new CommandResultText("New Itinerary Created");
+    public CommandResultText execute(Model model) throws DukeException, FileNotFoundException {
```
Comment (Jefferson111) [inline src/main/java/duke/logic/commands/NewItineraryCommand.java]: ```this ``` is unnecessary

File: src/main/java/duke/logic/parsers/ParserUtil.java (hunk 7 above)
Comment (Sukrut1881) [reply src/main/java/duke/logic/parsers/ParserUtil.java]: Will do this after fixing my command bugs. Created and issue for 1.4

**Pull request merged**